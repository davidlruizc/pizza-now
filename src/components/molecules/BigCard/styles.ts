import { Card, CardBody, Row } from 'reactstrap';
import styled from 'styled-components';

export const BodyCard = styled(CardBody)<{ padding?: boolean; height?: string }>`
  padding: ${(props) => (props.padding ? '1.5rem' : '0')} !important;
  width: 100%;
  height: ${(props) => (props.height ? props.height : '100%')};
`;

export const CardContainer = styled(Card)`
  margin-top: 1em;
  color: black;
`;

export const SubTitle = styled.p`
  margin-bottom: 0;
  opacity: 0.7;
  font-size: 0.8em;
`;

export const CardRow = styled(Row)``;
