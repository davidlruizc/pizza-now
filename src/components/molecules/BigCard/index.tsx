import * as React from 'react';
import { Col } from 'reactstrap';
import { BodyCard, CardContainer, CardRow, SubTitle } from './styles';

interface BigCardProps {
  children: React.ReactNode;
  title: string;
  padding?: string;
  height?: string;
  subtitle?: string;
  actionButton?: React.ReactNode;
}

const BigCard: React.FC<BigCardProps> = ({
  children,
  title,
  padding,
  height,
  subtitle,
  actionButton,
}) => {
  return (
    <CardContainer className="shadow bg-secondary">
      <div className="bg-white border-0 card-header">
        <CardRow>
          <Col sm="12">
            <h2 className={'mb-0'}>{title}</h2>
            {subtitle && <SubTitle>{subtitle}</SubTitle>}
          </Col>
          {actionButton && (
            <Col xs={4} md={4} sm={4}>
              {actionButton}
            </Col>
          )}
        </CardRow>
      </div>
      <BodyCard padding={padding} height={height}>
        {children}
      </BodyCard>
    </CardContainer>
  );
};

export default BigCard;
