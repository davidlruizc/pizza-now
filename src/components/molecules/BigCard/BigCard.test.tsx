import React from 'react';
import BigCard from './index';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

// TODO: change test for children prop
describe('<BigCard />', () => {
  test('render big card', () => {
    const data = {
      title: 'Crea tu pizza',
      subtitle: 'Elige los sabores que más te gusten y arma una pizza a tu medida',
    };

    const component = render(
      <BigCard title={data.title} subtitle={data.subtitle}>
        Pizza Now
      </BigCard>
    );

    component.getByText('Crea tu pizza');
    component.getByText('Elige los sabores que más te gusten y arma una pizza a tu medida');

    const tree = renderer
      .create(
        <BigCard title={data.title} subtitle={data.subtitle}>
          Pizza Now
        </BigCard>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
