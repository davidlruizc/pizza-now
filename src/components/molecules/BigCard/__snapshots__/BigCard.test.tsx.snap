// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigCard /> render big card 1`] = `
<div
  className="sc-gtsrHT jLYIyH shadow bg-secondary card"
>
  <div
    className="bg-white border-0 card-header"
  >
    <div
      className="sc-hKFxyN row"
    >
      <div
        className="col-sm-12"
      >
        <h2
          className="mb-0"
        >
          Crea tu pizza
        </h2>
        <p
          className="sc-dlnjwi jozUwS"
        >
          Elige los sabores que m√°s te gusten y arma una pizza a tu medida
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-bdnxRM hXVjEJ card-body"
  >
    Pizza Now
  </div>
</div>
`;
